eng/verify-nupkgs.ps1

```diff
@@ -112,4 +116,50 @@ function Unzip {
     [System.IO.Compression.ZipFile]::ExtractToDirectory($zipfile, $outpath)
 }
 
+function Match-VersionAgainstBranch {
+    param ([string]$vsTestVersion, [string]$branchName)
+
+    # Output useful info.
+    Write-Host "VSTest Product Version: `"$vsTestVersion`""
+    Write-Host "Current Branch: `"$branchName`""
+
+    $versionIsRTM = $vsTestVersion -match "^\d+\.\d+\.\d+$"
+    $versionIsRelease = $vsTestVersion -match "^\d+\.\d+\.\d+\-release\-\d{8}\-\d{2}$"
+    $versionIsPreview = $vsTestVersion -match "^\d+\.\d+\.\d+\-preview\-\d{8}\-\d{2}$"
+
+    $isReleaseBranch = $branchName -like "rel/*"
+    $isPreviewBranch = $branchName -like "main"
+
+    if (!$isReleaseBranch -and !$isPreviewBranch)
+    {
+        Write-Host "Skipping check since branch is neither `"release`" nor `"preview`""
+        return
+    }
+
+    Write-Host "Matching branch against product version ... "
+    if ($isReleaseBranch -and !$versionIsRTM -and !$versionIsRelease)
+    {
+        Write-Error "Release version `"$vsTestVersion`" should either be RTM, or contain a `"release`" suffix."
+        Exit 1
+    }
+    if ($isPreviewBranch -and !$versionIsPreview)
+    {
+        Write-Error "Preview version `"$vsTestVersion`" should contain a `"preview`" suffix."
+        Exit 2
+    }
+
+    Write-Host "Branch and version check was successful."
+    Exit 0
+}
+
+function Verify-Version {
+    param ([string]$nugetDir)
+
+    $vsTestExe = [IO.Path]::Combine($nugetDir, 'tools', 'net462', 'Common7', 'IDE', 'Extensions', 'TestPlatform', 'vstest.console.exe')
```

Comment:
just so you know: This is one way to do it. another is just using the path. Powershell is very tolerant of / and \ and double or triple /// , and it does not care about backslash or forward slash on linux.
```suggestion
    $vsTestExe = "$nugetDir/tools/net462/Common7/IDE/Extensions/TestPlatform/vstest.console.exe"
```